cores/esp32/esp32-hal-cpu.c

```diff
@@ -37,54 +38,110 @@ const uint32_t MHZ = 1000000;
 static apb_change_t * apb_change_callbacks = NULL;
 static xSemaphoreHandle apb_change_lock = NULL;
 
+void IRAM_ATTR twiddle( const char* pattern){
+int len=strlen(pattern),pos=0;
+digitalWrite(D_A,HIGH);
+digitalWrite(D_B,LOW);
+while(pos<len){
+  if(pattern[pos++]=='1'){
+    digitalWrite(D_B,HIGH);
+  } else {
+    digitalWrite(D_B,LOW);
+  }
+  digitalWrite(D_B,LOW);
+}
+digitalWrite(D_A,LOW);
+digitalWrite(D_B,LOW);
+}  
+
+
 static void initApbChangeCallback(){
     static volatile bool initialized = false;
     if(!initialized){
-        initialized = true;
```

Comment:
it was like that on purpose if two threads call it at the same time. please revert :)